<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PublishDialog</class>
 <widget class="QDialog" name="PublishDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>481</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Publish</string>
  </property>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>79</y>
     <width>191</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>QoS Settings</string>
   </property>
   <property name="textFormat">
    <enum>Qt::AutoText</enum>
   </property>
  </widget>
  <widget class="QDialogButtonBox" name="button_OkCancel">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>640</y>
     <width>166</width>
     <height>25</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QFrame" name="frame">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>461</width>
     <height>521</height>
    </rect>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QComboBox" name="comboBox_durability">
    <property name="geometry">
     <rect>
      <x>154</x>
      <y>42</y>
      <width>162</width>
      <height>21</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>VOLATILE</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TRANSIENT_LOCAL</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>430</y>
      <width>201</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Lifespan</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>106</y>
      <width>441</width>
      <height>101</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>211</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Lease Duration (ms):</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_leaseDuration">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>40</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>INF</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_liveliness">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>10</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>AUTOMATIC</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>MANUAL_BY_PARTICIPANT</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>MANUAL_BY_TOPIC</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>171</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Kind:</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>221</width>
       <height>19</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Announcement Period (ms):&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_announcementPeriod">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>70</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>INF</string>
     </property>
    </widget>
    <zorder>label_9</zorder>
    <zorder>label_8</zorder>
    <zorder>label_16</zorder>
    <zorder>lineEdit_leaseDuration</zorder>
    <zorder>comboBox_liveliness</zorder>
    <zorder>lineEdit_announcementPeriod</zorder>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>191</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Liveliness</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="spin_HistoryQos">
    <property name="geometry">
     <rect>
      <x>154</x>
      <y>10</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QFrame" name="frame_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>369</y>
      <width>441</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>50</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLineEdit" name="lineEdit_Deadline">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>10</y>
       <width>301</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>INF</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>121</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Duration (ms):&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <zorder>label_19</zorder>
    <zorder>lineEdit_Deadline</zorder>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>181</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Ownership</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QFrame" name="frame_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>441</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>50</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLineEdit" name="lineEdit_Lifespan">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>10</y>
       <width>301</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>INF</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>121</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Duration (ms):&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <zorder>label_18</zorder>
    <zorder>lineEdit_Lifespan</zorder>
   </widget>
   <widget class="QCheckBox" name="checkBox_reliable">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>10</y>
      <width>111</width>
      <height>26</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Reliable</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>340</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Deadline</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>History:</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>42</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>13</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Durability:</string>
    </property>
    <property name="textFormat">
     <enum>Qt::AutoText</enum>
    </property>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>441</width>
      <height>71</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>141</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Kind:</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>41</y>
       <width>221</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Ownership Strength:</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="spin_ownershipStrength">
     <property name="geometry">
      <rect>
       <x>232</x>
       <y>41</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_ownership">
     <property name="geometry">
      <rect>
       <x>232</x>
       <y>10</y>
       <width>199</width>
       <height>21</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>SHARED_OWNERSHIP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>EXCLUSIVE_OWNERSHIP</string>
      </property>
     </item>
    </widget>
    <zorder>label_12</zorder>
    <zorder>label_11</zorder>
    <zorder>spin_ownershipStrength</zorder>
    <zorder>comboBox_ownership</zorder>
   </widget>
   <zorder>label_6</zorder>
   <zorder>label_10</zorder>
   <zorder>spin_HistoryQos</zorder>
   <zorder>checkBox_reliable</zorder>
   <zorder>label_7</zorder>
   <zorder>frame_2</zorder>
   <zorder>comboBox_durability</zorder>
   <zorder>label_13</zorder>
   <zorder>frame_3</zorder>
   <zorder>frame_4</zorder>
   <zorder>label_14</zorder>
   <zorder>frame_5</zorder>
   <zorder>label_15</zorder>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>9</y>
     <width>413</width>
     <height>57</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <widget class="QLabel" name="label">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="label_3">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Size</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="3" colspan="2">
     <widget class="QLabel" name="label_4">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Partition</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QComboBox" name="combo_Shape">
      <item>
       <property name="text">
        <string>Square</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Triangle</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Circle</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QComboBox" name="combo_Color">
      <item>
       <property name="text">
        <string>RED</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>BLUE</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>GREEN</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>YELLOW</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>ORANGE</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>CYAN</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>MAGENTA</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>PURPLE</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>GRAY</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>BLACK</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QSpinBox" name="spin_Size">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="value">
       <number>30</number>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QCheckBox" name="checkBox_A">
      <property name="text">
       <string>A</string>
      </property>
     </widget>
    </item>
    <item row="1" column="4">
     <widget class="QCheckBox" name="checkBox_B">
      <property name="text">
       <string>B</string>
      </property>
     </widget>
    </item>
    <item row="1" column="5">
     <widget class="QCheckBox" name="checkBox_C">
      <property name="text">
       <string>C</string>
      </property>
     </widget>
    </item>
    <item row="1" column="6">
     <widget class="QCheckBox" name="checkBox_D">
      <property name="text">
       <string>D</string>
      </property>
     </widget>
    </item>
    <item row="0" column="5">
     <widget class="QCheckBox" name="checkBox_Asterisk">
      <property name="text">
       <string>*</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label_2">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Shape</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>combo_Shape</tabstop>
  <tabstop>combo_Color</tabstop>
  <tabstop>spin_Size</tabstop>
  <tabstop>checkBox_A</tabstop>
  <tabstop>checkBox_B</tabstop>
  <tabstop>checkBox_C</tabstop>
  <tabstop>checkBox_D</tabstop>
  <tabstop>checkBox_Asterisk</tabstop>
  <tabstop>spin_HistoryQos</tabstop>
  <tabstop>checkBox_reliable</tabstop>
  <tabstop>comboBox_durability</tabstop>
  <tabstop>comboBox_liveliness</tabstop>
  <tabstop>lineEdit_leaseDuration</tabstop>
  <tabstop>lineEdit_announcementPeriod</tabstop>
  <tabstop>comboBox_ownership</tabstop>
  <tabstop>spin_ownershipStrength</tabstop>
  <tabstop>lineEdit_Deadline</tabstop>
  <tabstop>lineEdit_Lifespan</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>button_OkCancel</sender>
   <signal>accepted()</signal>
   <receiver>PublishDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_OkCancel</sender>
   <signal>rejected()</signal>
   <receiver>PublishDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
